packages/api/src/controllers/sensorController.js

```diff
@@ -139,17 +179,49 @@ const sensorController = {
         );
 
         if (success.length + already_owned.length < esids.length) {
-          res.status(400).send({
-            error_type: 'unable_to_claim_all_sensors',
-            registeredSensors,
-          });
+          return hasTimedOut
+            ? await sendSensorNotification(
+                user_id,
+                farm_id,
+                SensorNotificationTypes.SENSOR_BULK_UPLOAD_FAIL,
+                {
+                  error_download: {
+                    errors: errorSensors,
+                    file_name: 'sensor-upload-outcomes.txt',
+                    is_validation_error: false,
+                    success,
+                  },
+                },
+              )
+            : res.status(400).send({
+                error_type: 'unable_to_claim_all_sensors',
+                success,
+                errorSensors,
+              }) && clearTimeout(timer);
         } else {
-          res.status(200).send({ message: 'Successfully uploaded!', sensors: sensorLocations });
+          return hasTimedOut
+            ? await sendSensorNotification(
+                user_id,
+                farm_id,
+                SensorNotificationTypes.SENSOR_BULK_UPLOAD_SUCCESS,
+              )
+            : res
+                .status(200)
+                .send({ message: 'Successfully uploaded!', sensors: sensorLocations }) &&
+                clearTimeout(timer);
         }
       }
     } catch (e) {
       console.log(e);
-      res.status(500).send(e.message);
+      const { user_id } = req.user;
```

Comment:
`user_id` and `farm_id` are already defined, just not in this scope. Move the definitions out of the `try` block and eliminate the repetition.

Reply:
Resolved